File: MasterApp/app/src/main/java/br/com/ysimplicity/masterapp/helper/SessionManager.java
package br.com.ysimplicity.masterapp.helper;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

/**
 * Created by Indigo on 11/18/16.
 */

public class SessionManager {

    private final static String AUTH_TOKEN = "AUTH_TOKEN";

    public static void saveToken(Context context, String token) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = prefs.edit();
        editor.putString(AUTH_TOKEN, token);
        editor.apply();
    }

    public static boolean isUserSignedIn(Context context) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);

        if (prefs.getString(AUTH_TOKEN, "").isEmpty()){
            return false;
        } else {
            return true;
        }
    }

    public static void logoutUser(Context context) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = prefs.edit();
        editor.clear();
        editor.apply();
    }


}

File: MasterApp/app/src/main/java/br/com/ysimplicity/masterapp/helper/SessionManager.kt
package br.com.ysimplicity.masterapp.helper

import android.content.Context
import android.preference.PreferenceManager

/**
 * Created by Ghost on 14/11/2017.
 */
class SessionManager {
    companion object {
        private val auth_token = "AUTH_TOKEN"

        fun saveToken(context: Context, token: String) {
            val prefs = PreferenceManager.getDefaultSharedPreferences(context)
            val editor = prefs.edit()
            editor.putString(auth_token, token)
            editor.apply()
        }

        fun isUserSignedIn(context: Context): Boolean {
            val prefs = PreferenceManager.getDefaultSharedPreferences(context)

            return prefs.getString(auth_token, "").isNotEmpty()
        }

        fun logoutUser(context: Context) {
            val prefs = PreferenceManager.getDefaultSharedPreferences(context)
            val editor = prefs.edit()
            editor.clear()
            editor.apply()
        }
    }
}